storage/metrics/metrics.go

```diff
@@ -0,0 +1,35 @@
+package metrics
```

Comment:
Would it be cleaner to have the types in this package live in the `storage` package?
```
storage.Metrics
storage.TSICacheMetrics
storage.WALMetrics
etc.
```

Reply:
I started with them there, but got cyclic import errors.

I'll see if they work out in the `tsdb` package (the 1.x storage tree). That feels like it could be a cleaner abstraction anyways, since new storage engines would probably(?) have different metrics.